### PR TITLE
Modify items() to iteritems()

### PR DESCRIPTION
**Pull request by lonelyandrew**
Use  `iteritems()`  to iterate a dict may be better than using `items()`, please read the reference which I add and make sure that my modification is correct.
I'm sorry about that I'm not familiar with the syntax of the reStructuredText(I often use markdown to write docs ), so please check my commit carefully.
By the way, thanks, this little book is helpful.

### CODE REVIEW

File: docs/readability/not_using_items_to_iterate_over_a_dictionary.rst [hunk 1]
```diff
@@ -18,19 +18,30 @@ The code below defines a for loop that iterates over a dictionary named ``d``. F
...
+    for key,val in d.iteritems():
         print "{} = {}".format(key, val)
 
+``items()`` may be another choice to iterate a dictionary, but in fact, ``iteritems()`` is better.
+
+In Python 2, Python ``items()`` built a real list of tuples and returned that. That could potentially take a lot of extra memory.
+
+Then, generators were introduced to the language in general, and that method was reimplemented as a iterator-generator method named ``iteritems()``.
+
+In Python 3,the ``iteritems()`` is gone, instead, the ``items()`` returns an iterator just like how ``iteritems()`` works in Python 2.
```
Comment (timofurrer) [inline docs/readability/not_using_items_to_iterate_over_a_dictionary.rst]: This is actually *not really true*. In python 3 `items` returns a `view object` to the items and not an iterator like the `iteritems` in python 2.7. 

See: 
https://docs.python.org/2/library/stdtypes.html#dict.iteritems
https://docs.python.org/3.3/library/stdtypes.html#dict.items

Comment (programmdesign) [inline docs/readability/not_using_items_to_iterate_over_a_dictionary.rst]: Agreed. @lonelyandrew, @timofurrer: would you mind updating the article and resolve merge conflicts so I can merge it finally (sorry it took so long)

**Comment by timofurrer** [comment]
I cannot change the pull request. It has to be done by @lonelyandrew . If he doesn't respond I pick his change on submit a new one. But let's wait a day more.

**Comment by programmdesign** [comment]
:+1:

**Comment by programmdesign** [comment]
Merged #85, which reflects this PR too.

**Pull request closed without merging**